Drivers/rc_receiver/Tests/sbus_gtest.cpp

```diff
@@ -0,0 +1,98 @@
+#include <gtest/gtest.h>
+#include <fff.h>
+#include "sbus_receiver.hpp"
+
+DEFINE_FFF_GLOBALS;
+
+/*
+    Faking the HAL function in this unit test to do nothing
+    The recieved data will be manually added to test cases
+*/
+FAKE_VALUE_FUNC(HAL_StatusTypeDef, HAL_UART_Receive_DMA, UART_HandleTypeDef*, uint8_t*, uint16_t );
+
+TEST (SBUS_Receiver_Test, RawDataToSBus){
+    const uint8_t in_bytes[SBUS_FRAME_SIZE] = {
+        HEADER_, 0x80, 0x08, 0x84,
+        0x20, 0x06, 0x41, 0x88, 0x42, 0x18,
+        0xE2, 0x10, 0x88, 0x48, 0x84, 0x22,
+        0x16, 0xC1, 0x88, 0x46, 0x38, 0xE2,
+        0x11, 0x00, FOOTER_
+    };
+
+    SBus_t packet{
+        packet.ch[0] = 128,
+        packet.ch[1] = 129,
+        packet.ch[2] = 130,
+        packet.ch[3] = 131,
+        packet.ch[4] = 132,
+        packet.ch[5] = 133,
+        packet.ch[6] = 134,
+        packet.ch[7] = 135,
+        packet.ch[8] = 136,
+        packet.ch[9] = 137,
+        packet.ch[10] = 138,
+        packet.ch[11] = 139,
+        packet.ch[12] = 140,
+        packet.ch[13] = 141,
+        packet.ch[14] = 142,
+        packet.ch[15] = 143
+    };
+
+    UART_HandleTypeDef uart_fake;
+    SBUSReceiver sbus_test(&uart_fake);
+    memcpy(sbus_test.raw_sbus_, in_bytes, sizeof(in_bytes));
+    sbus_test.parse();
+
+    for(int i = 0; i < SBUS_INPUT_CHANNELS; i ++){
+        EXPECT_TRUE(sbus_test.GetSBUS().ch[i] == packet.ch[i]);
+    }
+
+    for(int i = 0; i < SBUS_INPUT_CHANNELS; i ++){
+        EXPECT_TRUE(sbus_test.GetRCControl().ControlSignals[i] == 0);
+    }
```

Comment:
I don't quite understand why these are different (Why is GetRCControl always zero?)

I think a code comment may be useful here (or maybe just more test cases would be sufficient)?